File: pl/network/port_scanner/port_scanner.pl
#!/usr/bin/perl
#
use strict;
use Socket;
use feature 'say';

my $host = $ARGV[0];
say "Scan host: $host";
my $myhouse = inet_aton($host);

socket(SOCK, PF_INET, SOCK_STREAM, getprotobyname('tcp') );
for(my $i=1 ; $i<0xffff ; ++$i){
	my $paddr = sockaddr_in( $i, $myhouse );

	if( connect(SOCK, $paddr) ){
		say "$i:+"
	}else{
		say "$i:-"
	}
}
exit;

File: py/utils/port_scanner/port_scanner.py
#!/usr/bin/python3

import sys
import socket
import time

def scan_port(host, port):
    scan = socket.socket()
    scan.settimeout(0.1)
    try:
        scan.connect( (host, port) )
    except socket.error:
        return False
    else:
        return True


if __name__ == "__main__" :
	ports = [20, 21, 22, 23, 42, 43, 53, 67, 69, 80, 443]
	host = sys.argv[1]

	for port in ports:
		if scan_port(host, port):
			print(port)
	exit()
